tests/pruners_tests/test_successive_halving.py

```diff
@@ -0,0 +1,184 @@
+import optuna
+
+
+def test_successive_halving_pruner_with_one_trial():
+    # type: () -> None
+
+    study = optuna.study.create_study()
+    trial = optuna.trial.Trial(study, study.storage.create_new_trial_id(study.study_id))
+    trial.report(1, 1)
+    pruner = optuna.pruners.SuccessiveHalvingPruner(
+        min_resource=1, reduction_factor=2, min_early_stopping_rate=0)
+
+    # A pruner is not activated at a first trial.
+    assert not pruner.prune(study.storage, study.study_id, trial.trial_id, step=1)
+
+
+def test_successive_halving_pruner_intermediate_values():
+    # type: () -> None
+
+    pruner = optuna.pruners.SuccessiveHalvingPruner(
+        min_resource=1, reduction_factor=2, min_early_stopping_rate=0)
+    study = optuna.study.create_study()
+
+    trial = optuna.trial.Trial(study, study.storage.create_new_trial_id(study.study_id))
+    trial.report(1, 1)
+
+    # A pruner is not activated at a first trial.
+    assert not pruner.prune(study.storage, study.study_id, trial.trial_id, step=1)
+
+    trial = optuna.trial.Trial(study, study.storage.create_new_trial_id(study.study_id))
+    # A pruner is not activated if a trial has no intermediate values.
+    assert not pruner.prune(study.storage, study.study_id, trial.trial_id, step=1)
+
+    trial.report(2, 1)
+    # A pruner is activated if a trial has an intermediate value.
+    assert pruner.prune(study.storage, study.study_id, trial.trial_id, step=1)
+
+
+def test_successive_halving_pruner_up_to_third_rung():
```

Comment:
With this method name, I feel as if `third` were a boundary value. How about renaming this method to `test_successive_halving_pruner_rung_check`?